src/dft/backends/cufft/backward.cpp

```diff
@@ -51,6 +55,16 @@ ONEMKL_EXPORT void compute_backward(descriptor_type &desc,
     auto commit = detail::checked_get_commit(desc);
     auto queue = commit->get_queue();
     auto plan = detail::get_bwd_plan(commit);
+    auto offsets = detail::get_offsets(commit);
+
+    if constexpr (std::is_floating_point_v<fwd<descriptor_type>>) {
+        offsets[0] *= 2; // offset is supplied in complex but we offset scalar pointer
+        if (offsets[1] % 2 != 0) {
+            throw oneapi::mkl::unimplemented(
+                "DFT", "compute_forward(desc, inout)",
+                "cuFFT requires offset (first value in strides) to be multiple of `sizeof(complex)`!");
```

Comment:
If the value needs to be a multiple of `sizeof(complex)`, then why are you doing mod 2?

Reply:
it needs to be a multiple of `sizeof(complex)/sizeof(scalar)`, which equals 2. Fixed misleading error message.